### PR TITLE
Redirect successfully canceled Subscriptions to the Purchase List

### LINKED ISSUE #49720: Other Site Subscriptions: redirect to subscription list after removal
When you delete a subscription to another site, we keep you on an empty page after it's been removed. Let's take folks back to the purchases tab after the subscription deletes.

membership-removal

### PR DESCRIPTION
**Pull request by jjchrisdiehl**
#### Changes proposed in this Pull Request

When a subscription is successfully canceled from the account purchases list the canceled subscription's page should redirect back to the Purchases list - **_BONUS_** - a green 'success' notice will appear in the upper right corner of the Purchases page to give a visual cue that the cancelation was successfully completed.

Alternatively, if the cancelation fails for some reason, the customer will remain on the Subscriptions purchase page and a red error notice will instruct them to Contact Support with a link to `https://wordpress.com/help/contact`

#### Testing instructions

**Testing set up:**

1. Properly sandbox the api, see PCYsg-lW4-p2 under Testing.
2. Go to P9LPZB-gd-p2 (edit page for password)
3. Sign up for a recurring payment, in this step you should see the sandbox Stripe cards on file as shown https://d.pr/i/NY5hww

**Test 1:**
* Now, go to https://wordpress.com/me/purchases and click on the new Subscription item, then cancel it.
* You should then be redirected back to the Purchases list with the following success notice

Success:

image

**Test 2:**
* Add a new subscription following step 3 above
* Go to the subscription's purchase page where you see the option to cancel it - **don't cancel it**.
* To trigger the failure notice, open React component tools in your dev tools window and follow the diagram below
* TLDR - > Search for the Subscription[Localized] component -> change the value for the stoppingStatus prop from false to "fail" (quotes, this is a string) and then press enter to trigger the notice

image

Fixes #49720

### CODE REVIEW

File: client/me/memberships/subscription.jsx [hunk 1]
```diff
@@ -11,114 +12,125 @@ import HeaderCake from 'calypso/components/header-cake';
...
+		if ( stoppingStatus === 'fail' ) {
+			// run is-error notice to contact support
+			dispatch(
+				errorNotice(
+					translate(
+						'There was a problem while stopping your subscription, please {{a}}{{strong}}contact support{{/strong}}{{/a}}.',
+						{
+							components: {
+								a: <a href={ CALYPSO_CONTACT } />,
+								br: <br />,
```
Comment (sirbrillig) [inline client/me/memberships/subscription.jsx]: I'm sure this was probably copied from somewhere but I don't think there's a `br` used in this string.

Comment (jjchrisdiehl) [reply client/me/memberships/subscription.jsx]: Agreed, it was copied and I wasn't quite sure if it mattered. I suppose for conciseness it would be good to remove.

File: client/me/memberships/subscription.jsx [hunk 2]
```diff
@@ -11,114 +12,124 @@ import HeaderCake from 'calypso/components/header-cake';
...
-						text={ translate( 'This subscription has been stopped. You will not be charged.' ) }
-					/>
-				) }
+	useEffect( () => {
+		if ( stoppingStatus === 'fail' ) {
+			// run is-error notice to contact support
+			dispatch(
+				errorNotice(
+					translate(
+						'There was a problem while stopping your subscription, please {{a}}{{strong}}contact support{{/strong}}{{/a}}.',
```
Comment (sirbrillig) [inline client/me/memberships/subscription.jsx:40]: This changes the translation, but since this is an error state it's probably not that common and I don't know if it's worth using `hasTranslation` (or a string freeze) in this case. I'll leave that decision to you.

**Review by sirbrillig — APPROVED** [review]
Code looks good to me. We tested it in a pairing session and it works well!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6578768

Thank you @jjchrisdiehl for including a screenshot in the description! This is really helpful for our translators.

File: client/me/memberships/subscription.jsx [hunk 3]
```diff
@@ -11,114 +12,124 @@ import HeaderCake from 'calypso/components/header-cake';
...
+							},
+						}
+					)
+				)
+			);
+		} else if ( stoppingStatus === 'success' ) {
+			// redirect back to Purchases list
+			dispatch(
+				successNotice(
+					translate( 'This subscription has been stopped. You will not be charged.' ),
```
Comment (chrisfromthelc) [inline client/me/memberships/subscription.jsx]: I find the use of "stopped" versus "cancelled" a bit odd. Since we use "cancel" everywhere else, should we make these unified with that existing standard? Also, I would consider clarifying that you would not be charged in the future; "You will not be charged" could be misconstrued as their *current* charge would be wiped out (which does not appear to be the case).

**Review by chrisfromthelc — APPROVED** [review]
I had a suggestion/question about some of the language, but otherwise this works for me!

File: client/me/memberships/subscription.jsx (hunk 3 above)
Comment (jjchrisdiehl) [reply client/me/memberships/subscription.jsx]: Good points, I can add this in as it does make it more clear. Since it is only used in this error it shouldn't be too big of an issue.

File: client/me/memberships/subscription.jsx (hunk 2 above)
Comment (jjchrisdiehl) [reply client/me/memberships/subscription.jsx:40]: True, though I did add the string freeze tag already. Is it easy enough to just remove it? Or has a cascading series of events already started?

Comment (sirbrillig) [reply client/me/memberships/subscription.jsx:40]: I honestly don't know.

File: client/me/memberships/subscription.jsx (hunk 3 above)
Comment (sirbrillig) [reply client/me/memberships/subscription.jsx]: Worth noting that this string should be much more common than the other one, so now that this one is changed we should definitely wait for the translations to complete before merging; or alternatively use `hasTranslation` to use the old string as a fallback.

Comment (jjchrisdiehl) [reply client/me/memberships/subscription.jsx]: @sirbrillig Can you elaborate on what you mean by 'this string _should be_ much more common than the other one?' 

The revised string would be 'This subscription has been canceled. You will no longer be charged.' once finalized.

Comment (sirbrillig) [reply client/me/memberships/subscription.jsx]: I just mean that this message is displayed when the cancelation is successful, which is the happy path that comes from using this page, so it's likely to be displayed more often than the failure message above. I commented above that we might not need to wait for the full translation of the failure message since it's not likely to be displayed very often, but I don't think that reasoning holds true for the translation of the success message.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6578768).
